compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1048,6 +1047,38 @@ object RefChecks {
 
   end checkUnaryMethods
 
+  /** Check that an extension method is not hidden, i.e., that it is callable.
+   *
+   *  An extension method is hidden if it does not offer a parameter that is not subsumed
+   *  by the corresponding parameter of the member (or of all alternatives of an overload).
+   *
+   *  If the member has implicit parameters, then the application must supply explicit `using`
+   *  arguments to select an extension method. The reason is that if the arguments are not
+   *  supplied implicitly, typechecking fails and stops. But with explicit arguments, extensions
+   *  may be considered, and of course the extension must also accept implicit parameters.
+   *
+   *  If the extension method is nullary, it is always hidden by a member of the same name.
+   */
+  def checkExtensionMethods(sym: Symbol)(using Context): Unit = if sym.is(Extension) then
+    extension (tp: Type)
+      def firstExplicitParamTypes = Applications.stripImplicit(tp.stripPoly, wildcardOnly = true).firstParamTypes
+      def hasImplicitParams = tp.stripPoly match { case mt: MethodType => mt.isImplicitMethod case _ => false }
+    val target = sym.info.firstParamTypes.head // required for extension method
+    if !target.typeSymbol.denot.isAliasType && !target.typeSymbol.denot.isOpaqueAlias then
+      val methTp = sym.denot.info.resultType
```

Comment:
That also looks like it will do the wrong thing for extensions starting with a using clause (also `sym.denot.info` is equivalent to `sym.info`), e.g.:
```scala
class C { def foo(x: Int): Int = 0 }
extension (using Int)(c: C) def foo(x: Int): Int = 1
```